tests/boards/intel_adsp/hda/src/smoke.c

```diff
@@ -0,0 +1,189 @@
+{
+}
+
+
+/**
+ * Tests host input streams
+ *
+ * Note that the order of operations in this test are important and things potentially will not
+ * work in horrible and unexpected ways if not done as they are here.
+ *
+ * Specifically configuring the host side buffers before the dsp side seems to cause things to
+ * become invalid.
+ */
+void test_hda_host_in_smoke(void)
+{
+	int res;
+	uint32_t last_msg_cnt;
+
+	printk("smoke testing hda with fifo buffer at address %p, size %d\n",
+	       hda_fifo, FIFO_SIZE);
+
+	cavs_ipc_set_message_handler(CAVS_HOST_DEV, ipc_message, NULL);
+	cavs_ipc_set_done_handler(CAVS_HOST_DEV, ipc_done, NULL);
+
+	printk("Using buffer of size %d at addr %p\n", FIFO_SIZE, hda_fifo);
+
+	/* setup a ramp in the buffer */
+	for (uint32_t i = 0; i < FIFO_SIZE; i++) {
+		hda_fifo[i] = i & 0xff;
+	}
+	/* The buffer is in the cached address range and must be flushed */
+	z_xtensa_cache_flush(hda_fifo, FIFO_SIZE);
+
+	cavs_hda_init(HDA_HOST_IN_BASE, STREAM_ID);
+	printk("dsp init: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+
+	WAIT_FOR(cavs_ipc_send_message_sync(CAVS_HOST_DEV, IPCCMD_HDA_RESET,
+					    STREAM_ID, IPC_TIMEOUT));
+	printk("host reset: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+
+	WAIT_FOR(cavs_ipc_send_message_sync(CAVS_HOST_DEV, IPCCMD_HDA_CONFIG,
+					    STREAM_ID | (FIFO_SIZE << 8),
+					    IPC_TIMEOUT));
+	printk("host config: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+
+
+	res = cavs_hda_set_buffer(HDA_HOST_IN_BASE, STREAM_ID, hda_fifo, FIFO_SIZE);
+	printk("dsp set_buffer: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+	zassert_ok(res, "Expected set buffer to succeed");
+
+	cavs_hda_enable(HDA_HOST_IN_BASE, STREAM_ID);
+	printk("dsp enable: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+
+	WAIT_FOR(cavs_ipc_send_message_sync(CAVS_HOST_DEV, IPCCMD_HDA_START,
+					    STREAM_ID, IPC_TIMEOUT));
+	printk("host start: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+
+	for (uint32_t i = 0; i < TRANSFER_COUNT; i++) {
+		cavs_hda_commit(HDA_HOST_IN_BASE, STREAM_ID, FIFO_SIZE);
+		printk("dsp inc_pos: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+
+		WAIT_FOR(cavs_hda_wp_rp_eq(HDA_HOST_IN_BASE, STREAM_ID));
+		printk("dsp wp_rp_eq: "); cavs_hda_dbg("host_in", HDA_HOST_IN_BASE, STREAM_ID);
+
+		k_sleep(K_MSEC(10));
+
+		last_msg_cnt = msg_cnt;
+		WAIT_FOR(cavs_ipc_send_message_sync(CAVS_HOST_DEV, IPCCMD_HDA_VALIDATE,
+						    STREAM_ID, IPC_TIMEOUT));
+		WAIT_FOR(msg_cnt > last_msg_cnt);
+		zassert_true(msg_res == 1,
+			     "Expected data validation to be true from Host");
+	}
+
+	WAIT_FOR(cavs_ipc_send_message_sync(CAVS_HOST_DEV, IPCCMD_HDA_RESET,
+					    STREAM_ID, IPC_TIMEOUT));
+	cavs_hda_disable(HDA_HOST_IN_BASE, STREAM_ID);
+}
+
+void test_hda_host_out_smoke(void)
+{
+	int res;
+	bool is_ramp;
+
+	printk("smoke testing hda with fifo buffer at address %p, size %d\n",
+	       hda_fifo, FIFO_SIZE);
```

Comment:
Maybe a little too much printk traffic here for my taste.  In test code no one's going to read it unless it fails, so it's generally best to put the logging into the zassert_*() cases (which have a printf format string and arguments at the end).

Reply:
For sure, was very helpful when I tested many, many times to see the screen blip by with my ramp data (0-255) or all zeros. Definitely overkill now that I have something working.